vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/features/metadata/source/exif/ExifMetadataExtractor.kt

```diff
@@ -65,24 +65,39 @@ private fun convertType(directory: Directory, tagType: Int, type: Type): Value<*
     is Type.LongVector -> throw IllegalArgumentException("Unsupported type: $type")
 }
 
-private fun JsonObject.convertType(type: Type): Value<*>? {
-    val jsonPrimitive = this.jsonPrimitive
-    if (jsonPrimitive.isString) {
-        return when (type) {
-            Type.String -> Value.String(jsonPrimitive.content)
-            Type.Datetime -> convertDate(jsonPrimitive.content)?.let { Value.DateTime(it) }
-            else -> null
+private fun JsonElement.convertType(type: Type): Value<*>? {
```

Comment:
I don't understand why this is built this way. The `JsonElement` is already typed, so why is there an explicit type parameter? Wouldn't it be more straightforward to just have a `JsonElement.toValue()` method that converts the element into a value object with the corresponding type?

Reply:
I would argue that those are two independent problems. If JSON cannot represent the type in anything else than a string, it will be a string. This string can then, in a subsequent step, be parsed into something else.

Reply:
No, what I am saying is that there should be a `fun JsonElement.toValue() : Value<*>` that does not need to be private, as it can be potentially useful in other places as well, followed by whatever local task-specific logic you might need to compensate for the lack of support for certain types in JSON.

Reply:
Ok, I have pushed a change. Is this more or less what you meant? Or did you mean a fully fledged value conversion system in core.model.types?

Reply:
This looks much more like what I had in mind. I don't think we need a complete value type conversion system for this. If we ever need one, we can build it then and replace the current functionality with it. Thoughts @ppanopticon?